### ISSUE TITLE
SC2203 assumes [] is always a globbing attempt, even though it can also be referencing an array

### ISSUE THREAD

**Issue by TheCayman** [koalaman/shellcheck#1696]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2203
- My shellcheck version (`shellcheck --version` or "online"): v0.7.0
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
declare -a arr=(1)
[[ arr[0] -gt 0 ]] && echo "Greater than 0"
```

#### Here's what shellcheck currently says:
Line 3:
[[ arr[0] -gt 0 ]] && echo "Greater than 0"
   ^-- SC2203: Globs are ignored in [[ ]] except right of =/!=. Use a loop.

#### Here's what I wanted or expected to see:
No warnings, as this is perfectly valid bash.

This may be related to issue 1641, as that deals with * being interpreted as globbing instead of multiplication causing SC2203 being raised.

**Comment by koalaman** [comment]
You're exactly right. This has been fixed along with #1641. Thanks!
